dex/farm-with-locked-rewards/src/lib.rs

```diff
@@ -100,6 +100,15 @@ pub trait Farm:
         (new_farm_token, boosted_rewards).into()
     }
 
+    #[endpoint(updateEnergyForUser)]
+    fn update_energy_for_user(&self, user: ManagedAddress) {
+        let current_week = self.get_current_week();
+        let claim_progress = self.current_claim_progress(&user).get();
+        if claim_progress.week == current_week {
```

Comment:
I think this should be an error instead. We want to know if something went wrong.